hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/Reactor.java

```diff
@@ -361,40 +364,44 @@ private StartEventloopTask(CompletableFuture<Eventloop> future, ReactorBuilder b
         @Override
         public void run() {
             try {
+                run0();
+            } catch (Exception e) {
+                // log whatever wasn't caught so that we don't swallow throwables.
+                logger.severe(e);
+            }
+        }
+
+        private void run0() {
```

Comment:
Why are you catching an exception if run0 can't throw an Exception.

Reply:
I'm not happy with it. Now the logic is being smeared over multiple methods.

Also, I want to keep the call stack at the bottom of the call stack as short as possible. It makes debugging simpler.